cylc/flow/scheduler.py

```diff
@@ -1645,6 +1645,7 @@ async def main_loop(self) -> None:
             # Shutdown workflow if timeouts have occurred
             self.timeout_check()
 
+            self.pool.spawn_children()
```

Comment:
I think a lot of the functional tests can be sensitive to the order of main loop events. Maybe try to relocate this where spawning happened before (process_queued_task_messages?).